### PR TITLE
Metaparser example

### PR DESCRIPTION
**Pull request by karmoham**

### CODE REVIEW

File: coding-201-advanced-parsers/metaparser.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
+'''
+Metaparser
+----------
+   
+   Metaparser pyats module provides a structure unification among different
+   parsing contexts. It provides a unified structure that will be followed
+   by different device connections (Ex: 'cli', 'yang', 'xml).
+
+   It addresses the common automation problems as:
```
Comment (simingy) [inline coding-201-advanced-parsers/metaparser.py]: It addresses common automation problems such as:

File: coding-201-advanced-parsers/metaparser.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        Neighbor: 21.0.0.4, Address-Family: VPNv6 Unicast (VRF2)
+           Locally configured policies:
+            route-map RMAP7 in
+            route-map RMAP8 out
+        '''
+
+    def cli(self):
+
+        # Initialize dictionary
+        PYTHON_DICT = {}
```
Comment (simingy) [inline coding-201-advanced-parsers/metaparser.py]: name this result

File: coding-201-advanced-parsers/metaparser.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                        }
+                    },
+                }
+            },
+        }
+
+
+class Parser(ParserSchema):
+
+    STRING = '''
```
Comment (simingy) [inline coding-201-advanced-parsers/metaparser.py]: maybe leave this as docstring header

File: coding-201-advanced-parsers/metaparser.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+#                     {'address_family':
+#                         {'VPNv4 Unicast':
+#                             {'route_map_in': 'RMAP5',
+#                              'route_map_out': 'RMAP6'}
+#                         }
+#                     }
+#                 }
+#             }
+#         }
+#     }
```
Comment (simingy) [inline coding-201-advanced-parsers/metaparser.py]: if __name__ = '__main__':
    # give an example of running it

**Review by simingy — CHANGES_REQUESTED** [review]
let's keep this as metaparser_schema.py

and add one for cli/rest, and showcase the power of metaparser

File: coding-201-advanced-parsers/metaparser.py (hunk 1 above)
Comment (karmoham) [reply coding-201-advanced-parsers/metaparser.py]: Addressed.

File: coding-201-advanced-parsers/metaparser.py (hunk 2 above)
Comment (karmoham) [reply coding-201-advanced-parsers/metaparser.py]: Done

File: coding-201-advanced-parsers/metaparser.py (hunk 4 above)
Comment (karmoham) [reply coding-201-advanced-parsers/metaparser.py]: Done.

File: coding-201-advanced-parsers/metaparser.py (hunk 3 above)
Comment (karmoham) [reply coding-201-advanced-parsers/metaparser.py]: If we need to have it as a running example, I can't have it as a docstring.

**Review by simingy — CHANGES_REQUESTED** [review]
you committed a .pyc file

**Review by simingy — APPROVED** [review]

**Pull request merged**